python/revng/scripting/projects/cli_project.py

```diff
@@ -0,0 +1,106 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import subprocess
+from tempfile import NamedTemporaryFile
+from typing import List
+
+import yaml
+
+from revng.model import Binary, DiffSet, Function  # type: ignore[attr-defined]
+
+from .project import Project
+
+
+class CLIProject(Project):
+    """
+    This class is used to run revng analysis and artifact through
+    the CLI with subprocess.
+    """
+
+    def __init__(self, resume_dir: str | None = None, revng_executable: str | None = None):
+        self._set_resume_dir(resume_dir)
+        self._set_revng_executable(revng_executable)
+        super().__init__()
+
+    def _load(self):
+        self._get_pipeline_description()
+        # TODO: We hardcode the path to the `model.yml`, fix when there is a
+        # command to get the model.
+        model_path = f"{self.resume_dir}/context/model.yml"
+        if not os.path.exists(model_path) and not os.path.isfile(model_path):
+            return
+        with open(model_path) as f:
+            self._set_model(Binary.deserialize(f))
+
+    def import_binary(self, input_binary_path: str):
```

Comment:
Bad name, `import-binary` is the name of an important analysis, but this is not it.

Reply:
I know, but how do you name the function that takes in the input binary **without** running the initial auto-analyses?

Reply:
`set_executable` or `set_executable_path`

Reply:
Implemented as `set_binary_path`